pandas/io/sql.py

```diff
@@ -2069,6 +2080,16 @@ def drop_table(self, table_name: str, schema: str | None = None) -> None:
                 self.get_table(table_name, schema).drop(bind=self.con)
             self.meta.clear()
 
+    def delete_rows(self, table_name: str, schema: str | None = None) -> None:
+        schema = schema or self.meta.schema
+        if self.has_table(table_name, schema):
```

Comment:
So in the case the table does not exist we are just ignoring any instruction to perform a DELETE? I'm somewhat wary of assuming a user may not want an error here

Reply:
Yeah, me neither.
We can remove this check letting the driver error and eventually raise a `DatabaseError`.
What you think ?

Reply:
On second though I think this is OK. It follows the same pattern as `replace` which will create the table if it does not exist